compendium/ServiceComponent/test/TestCompInst_IncorrectActivateMethod.cpp

```diff
@@ -0,0 +1,74 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  =============================================================================*/
+
+#include "gtest/gtest.h"
+
+#include "cppmicroservices/servicecomponent/ComponentContext.hpp"
+#include "cppmicroservices/servicecomponent/detail/ComponentInstanceImpl.hpp"
+#include "cppmicroservices/servicecomponent/usrDefinedMethodAssertion.hpp"
+#include <cppmicroservices/ServiceInterface.h>
+
+using cppmicroservices::service::component::detail::ComponentInstanceImpl;
+using ComponentContext = cppmicroservices::service::component::ComponentContext;
+
+namespace
+{
+    struct TestServiceInterface1
+    {
+    };
+
+    class ServiceComponentWrongActivateSig final
```

Comment:
Would names for these base classes be more readable and intuitive if they were something like: ServiceComponentCheckActivateSignature??